File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAscendingSum = function (nums) {
    if (nums.length == 1) return nums[0]
    let sum = nums[0]
    let arr = []
    for (let i = 1; i < nums.length; i++) {
        if (nums[i - 1] < nums[i]) {
            sum += nums[i]
        }
        else {
            arr.push(sum)
            sum = nums[i]
        }
        if (i == nums.length - 1) {
            arr.push(sum)
        }
    }
    return Math.max(...arr)
};

File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        
        sum_ = nums[0]
        arr = []
        
        for i in range(1, len(nums)):
            if nums[i - 1] < nums[i]:
                sum_ += nums[i]
            else:
                arr.append(sum_)
                sum_ = nums[i]
            
            if i == len(nums) - 1:
                arr.append(sum_)
        
        return max(arr)